lua/null-ls/builtins/diagnostics/commitlint.lua

```diff
@@ -0,0 +1,67 @@
+local h = require("null-ls.helpers")
+local methods = require("null-ls.methods")
+
+local DIAGNOSTICS = methods.internal.DIAGNOSTICS
+
+local violation_to_diagnostic = function(violation)
+    local diagnostic = nil
+
+    diagnostic = {
+        ruleId = violation.name,
+        message = violation.message,
+        level = violation.level,
+    }
+
+    if violation.name == "body-leading-blank" then
+        diagnostic = vim.tbl_extend("force", diagnostic, { line = 2 })
+    elseif vim.startswith(violation.name, "body") then
+        diagnostic = vim.tbl_extend("force", diagnostic, { line = 3 })
+    end
+
+    return diagnostic
+end
+
+local handle_commitlint_output = function(params)
+    if params.output and params.output.results then
+        local output = params.output.results[1]
+
+        local parser = h.diagnostics.from_json({
+            severities = {
+                [1] = h.diagnostics.severities.warning,
+                [2] = h.diagnostics.severities.error,
+            },
+        })
+
+        local violations = {}
+        for _, violation in ipairs(output.errors) do
+            table.insert(violations, violation_to_diagnostic(violation))
+        end
+
+        for _, violation in ipairs(output.warnings) do
+            table.insert(violations, violation_to_diagnostic(violation))
+        end
+
+        return parser({ output = violations })
+    end
+end
+
+return h.make_builtin({
+    name = "commitlint",
+    meta = {
+        url = "https://commitlint.js.org",
+        description = "commitlint checks if your commit messages meet the conventional commit format.",
+    },
+    method = DIAGNOSTICS,
+    filetypes = { "gitcommit" },
+    generator_opts = {
+        command = "commitlint",
+        args = { "--format", "commitlint-format-json" },
+        to_stdin = true,
+        format = "json_raw",
```

Comment:
```suggestion
        format = "json",
```

We should only use `json_raw` if we are planning on handling errors ourselves (in this case, this suppressed an error I was getting from the CLI, making it harder to figure out what was going on).

Reply:
I initially switched it to `json_raw` because when I was typing a commit, then backspaced it to an empty commit, it would throw a JSON parsing error. But I can't even reproduce that now, so likely I was doing something else wrong. It does make more sense to let the errors bubble up. I got this changed! I appreciate your help.